components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAOTest.java

```diff
@@ -0,0 +1,1486 @@
+                assertEquals(idpFromDB.getIdentityProviderName(), idpName,
+                        "'getIDPbyResourceId' method fails");
+                if (idpFromCache != null) {
+                    assertEquals(idpFromCache.getIdentityProviderName(), idpName,
+                            "'getIDPbyResourceId' method fails");
+                }
+            } else {
+                assertNull(idpResults);
+            }
+        }
+    }
+
+    @DataProvider
+    public Object[][] getIDPNameByResourceIdData() {
+
+        return new Object[][]{
+                {"testIdP1", SAMPLE_TENANT_ID},
+                {"testIdP3", SAMPLE_TENANT_ID2}
+        };
+    }
+
+    @Test(dataProvider = "getIDPNameByResourceIdData")
+    public void testGetIdPNameByResourceId(String idpName, int tenantId) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDataSource()).thenReturn(dataSourceMap.get(DB_NAME));
+            addTestIdps();
+
+            String uuid = "";
+            // getting IDP from DB
+            IdentityProvider idPResult = idPManagementDAO.getIdPByName(connection, idpName, tenantId, TENANT_DOMAIN);
+            uuid = idPResult.getResourceId();
+            // getting IDP using resourceId from DB and adding to cache
+            IdentityProvider idpFromDB = cacheBackedIdPMgtDAO.getIdPByResourceId(uuid, tenantId, TENANT_DOMAIN);
+            IdentityProvider idpFromCache = idpFromCacheByResourceId(uuid);
+
+            mockStatic(CarbonContext.class);
+            CarbonContext carbonContext = mock(CarbonContext.class);
+            when(CarbonContext.getThreadLocalCarbonContext()).thenReturn(carbonContext);
+            when(CarbonContext.getThreadLocalCarbonContext().getTenantDomain()).thenReturn(TENANT_DOMAIN);
+            // retrieving IDP name from cache using resourceId as cache key
+            String name = cacheBackedIdPMgtDAO.getIdPNameByResourceId(uuid);
+
+            if (idpFromCache != null) {
+                assertEquals(idpFromCache.getIdentityProviderName(), idpName,
+                        "'getIDPNameByResourceId' method fails");
+                assertEquals(idpFromCache.getIdentityProviderName(), idpName,
+                        "'getIDPNameByResourceId' method fails");
+            }
+            assertEquals(name, idpName);
+        }
+    }
+
+    @DataProvider
+    public Object[][] getIdPByAuthenticatorPropertyValueWithoutAuthenticatorData() {
+
+        return new Object[][]{
+                {SAMPLE_TENANT_ID, "testIdP1", "Property1", "value1", true},
+                {SAMPLE_TENANT_ID, "testIdP1", "Property2", "value2", true},
+                {SAMPLE_TENANT_ID, "NotExist", "Null", "Null", false},
```

Comment:
Same value for SAMPLE_TENANT_ID passed in all combination

Reply:
Check other places as well

Reply:
Resolved with https://github.com/wso2/carbon-identity-framework/pull/4110/commits/28a1194ed9f5de659bd1879f625cd070efdac3fa